runtime/bcverify/bcverify.c

```diff
@@ -107,8 +107,19 @@ setInitializedThisStatus(J9BytecodeVerificationData *verifyData)
 		if (currentStack->stackBaseIndex != -1) {
 			BOOLEAN flag_uninitialized = FALSE;
 			IDATA i = 0;
-			for (; i < currentStack->stackTopIndex; i++) {
-				if ((currentStack->stackElements[i] & BCV_SPECIAL_INIT) == BCV_SPECIAL_INIT) {
+			/* The operation stack frame in the Verifier is as follows:
+			 * |-------- Locals --------|-------- Stacks ----------|
+			 * 0                        stackBaseIndex             stackTopIndex
+			 * which means 'Locals' starts from 0 to stackBaseIndex while 'Stacks' starts from stackBaseIndex
+			 * to stackTopIndex.
+			 *
+			 * The JVM Spec at 4.10.1.4 Stack Map Frame Representation says that if any local variable in 'Locals'
+			 * has the type 'uninitializedThis', then 'Flags' has the single element 'flagThisUninit'; otherwise
+			 * 'Flags' is an empty list. That is to say, we only check the elements of 'Locals' rather than the
+			 * whole operation stack frame to determine whether to set the flags.
+			 */
+			for (; i < currentStack->stackBaseIndex; i++) {
+				if (J9_ARE_ALL_BITS_SET(currentStack->stackElements[i], BCV_SPECIAL_INIT)) {
```

Comment:
This looks right and matches the spec.  Does it pass all the internal tests and the compliance tests from JDK 8. 11, 17?

Reply:
> Does it pass all the internal tests and the compliance tests from JDK 8. 11, 17

It passed the compliance tests from JDK 8 but I didn't do that in Java 11 & 17.

Reply:
I will do the compliance tests on Java 17 as suggested by Dan.

Reply:
I already finished the compliance tests on Java 17 without any issue.